libpod/pod.go

```diff
@@ -209,6 +229,27 @@ func (p *Pod) CgroupPath() (string, error) {
 		return "", err
 	}
 
+	if !p.HasInfraContainer() {
```

Comment:
We should check `p.state.CgroupPath` here first, and if it's set, we can return it immediately.